static/app/utils/useResizable.tsx

```diff
@@ -0,0 +1,158 @@
+import type {RefObject} from 'react';
+import {useCallback, useEffect, useRef, useState} from 'react';
+
+interface UseResizableOptions {
+  /**
+   * The starting size of the container, and the size that is set in the onDoubleClick handler.
+   *
+   * If `sizeStorageKey` is provided and exists in local storage,
+   * then this will be ignored in favor of the size stored in local storage.
+   */
+  initialSize: number;
+  /**
+   * The maximum width the container can be resized to. Defaults to Infinity.
+   */
+  maxWidth: number;
+  /**
+   * The minimum width the container can be resized to. Defaults to 100.
+   */
+  minWidth: number;
+  /**
+   * The ref to the element to be resized.
+   */
+  ref: RefObject<HTMLElement | null>;
+  /**
+   * Triggered when the user finishes dragging the resize handle.
+   */
+  onResizeEnd?: (newWidth: number) => void;
+  /**
+   * Triggered when the user starts dragging the resize handle.
+   */
+  onResizeStart?: () => void;
+  /**
+   * The local storage key used to persist the size of the container. If not provided,
+   * the size will not be persisted and the defaultWidth will be used.
+   */
+  sizeStorageKey?: string;
+}
+
+interface UseResizableResult {
+  /**
+   * Whether the drag handle is held.
+   */
+  isHeld: boolean;
+  /**
+   * Apply this to the drag handle element to include 'reset' functionality.
+   */
+  onDoubleClick: () => void;
+  /**
+   * Attach this to the drag handle element's onMouseDown handler.
+   */
+  onMouseDown: (e: React.MouseEvent) => void;
```

Comment:
react aria kinda does these `buttonProps` it might make sense to have a `handleProps` if we think they should be applied in a uniform way